craft_parts/plugins/python_plugin.py

```diff
@@ -154,44 +154,91 @@ def get_build_commands(self) -> List[str]:
                 f"""\
-            determine_link_target() {{
-                opts_state="$(set +o +x | grep xtrace)"
-                interp_dir="$(dirname "${{PARTS_PYTHON_VENV_INTERP_PATH}}")"
-                # Determine python based on PATH, then resolve it, e.g:
-                # (1) <application venv dir>/bin/python3 -> /usr/bin/python3.8
-                # (2) /usr/bin/python3 -> /usr/bin/python3.8
-                # (3) /root/stage/python3 -> /root/stage/python3.8
-                # (4) /root/parts/<part>/install/usr/bin/python3 -> /root/parts/<part>/install/usr/bin/python3.8
-                python_path="$(which "${{PARTS_PYTHON_INTERPRETER}}")"
-                python_path="$(readlink -e "${{python_path}}")"
-                for dir in "{self._part_info.part_install_dir}" "{self._part_info.stage_dir}"; do
-                    if  echo "${{python_path}}" | grep -q "${{dir}}"; then
-                        python_path="$(realpath --strip --relative-to="${{interp_dir}}" \\
-                                "${{python_path}}")"
-                        break
+                # look for a provisioned python interpreter
+                opts_state="$(set +o|grep errexit)"
+                set +e
+                install_dir="{self._part_info.part_install_dir}/usr/bin"
+                stage_dir="{self._part_info.stage_dir}/usr/bin"
+                payload_python=$(find "$install_dir" "$stage_dir" -type f -executable -name "python3.*" -print -quit 2>/dev/null)
+
+                if [ -n "$payload_python" ]; then
+                    # We found a provisioned interpreter, use it.
+                    installed_python="${{payload_python##{self._part_info.part_install_dir}}}"
+                    if [ "$installed_python" = "$payload_python" ]; then
+                        # Found a staged interpreter.
+                        symlink_target="..${{payload_python##{self._part_info.stage_dir}}}"
+                    else
+                        # The interpreter was installed but not staged yet.
+                        symlink_target="..$installed_python"
                     fi
-                done
-                echo "${{python_path}}"
-                eval "${{opts_state}}"
-            }}
+                else
+                    # Otherwise use what _get_system_python_interpreter() told us.
+                    symlink_target="{python_interpreter}"
+                fi
+
+                if [ -z "$symlink_target" ]; then
+                    echo "No suitable Python interpreter found, giving up."
+                    exit 1
+                fi
 
-            python_path="$(determine_link_target)"
-            ln -sf "${{python_path}}" "${{PARTS_PYTHON_VENV_INTERP_PATH}}"
-            """
+                eval "${{opts_state}}"
+                ln -sf "${{symlink_target}}" "${{PARTS_PYTHON_VENV_INTERP_PATH}}"
+                """
             )
         )
 
         return build_commands
+
+    def _should_remove_symlinks(self) -> bool:
```

Comment:
What's the motivation behind this being a method rather than an attribute or property?

Reply:
This allows application-specific subclasses to use complex logic based on part info to decide what to do (instead of a simple yes or no),